### DISCUSSION TITLE
Where does table AzureOptimizationConsumptionV1_CL come from?

### DISCUSSION THREAD

**Discussion by feiranjiang** [microsoft/finops-toolkit#993] [Q&A] [answered]
In the workbook of costs-growing, it tries retrieving data from a kusto table 'AzureOptimizationConsumptionV1_CL'. However the workbook can't get this table, so i would like to know where this table resides and what action i should take?

**Comment by psilantropy** [comment]
I have the same or  similar issue. AzureOptimizationAADObjectsV1_CL table doesn't exist in my workspace.
Just logged issue here: https://github.com/microsoft/finops-toolkit/issues/995

**Comment by helderpinto** [comment, marked as answer]
@feiranjiang, sorry for the late reply. The `AzureOptimizationConsumptionV1_CL` table is fed by a couple of runbooks:

1. `Export-ConsumptionToBlobStorage` - this runbook collects Azure consumption data from the Cost Management APIs and exports it to blob storage (`consumptionexports` container in the AOE storage account).
2. `Ingest-OptimizationCSVExportsToLogAnalytics` - this is a generic runbook which reads CSV data from any AOE blob storage container and ingests it into a corresponding table in the Log Analytics workspace. There is a `AzureOptimization_IngestConsumptionDaily` schedule linked to this runbook, at which new CSV data present in the `consumptionexports` container is ingested.

If the `AzureOptimizationConsumptionV1_CL` table is still empty, I advise you to check for job errors in the runbooks above. Job errors are displayed in the "Exception" tab.
